compiler/modules/front.syntax/scan.bal

```diff
@@ -399,24 +447,49 @@ function scanNormal(int[] codePoints, int startIndex, Scanned result) {
                 i = scanDecimal(codePoints, i, result);
             }
             CP_PERCENT
-            |CP_AMPERSAND
             |CP_LEFT_PAREN 
             |CP_RIGHT_PAREN
-            |CP_ASTERISK
-            |CP_PLUS
             |CP_COMMA
-            |CP_MINUS
             |CP_COLON
             |CP_SEMICOLON 
             |CP_QUESTION
             |CP_LEFT_SQUARE
             |CP_RIGHT_SQUARE
-            |CP_CIRCUMFLEX
             |CP_RIGHT_CURLY
             |CP_TILDE => {
                 // JBUG when FragCode is byte, error without cast
                 endFragment(<FragCode>cp, i, result);
             }
+            CP_CIRCUMFLEX => {
+                if endFragmentCompoundAssign(codePoints, i, CP_CIRCUMFLEX, FRAG_CIRCUMFLEX_EQUAL, result) {
```

Comment:
This function should return the new value of i. Then instead of e.g.

```
 if endFragmentCompoundAssign(codePoints, i, CP_PLUS, FRAG_PLUS_EQUAL, result) {
                    i += 1;
                    continue;
                }
```

do

```
 i = endFragmentCompoundAssign(codePoints, i, CP_PLUS, FRAG_PLUS_EQUAL, result);
```

Reply:
fixed